x/appchain/coordinator/keeper/ibc_client.go

```diff
@@ -0,0 +1,251 @@
+	if err != nil {
+		return err
+	}
+	k.SetClientForChain(ctx, chainID, clientID)
+
+	epochInfo, _ := k.epochsKeeper.GetEpochInfo(ctx, req.EpochIdentifier)
+	initTimeoutPeriod := coordinatorParams.InitTimeoutPeriod
+	// the CurrentEpoch below is the one that has already ended, so we increment it by 1.
+	// assume we start with a value of 2 and we are giving 4 full epochs for initialization.
+	// so when epoch 6 ends, the timeout ends.
+	initTimeoutPeriod.EpochNumber += uint64(epochInfo.CurrentEpoch) + 1
+	k.AppendChainToInitTimeout(ctx, initTimeoutPeriod, chainID)
+
+	k.Logger(ctx).Info(
+		"subscriber chain registered (client created)",
+		"chainId", chainID,
+		"clientId", clientID,
+	)
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeSubscriberClientCreated,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+			sdk.NewAttribute(commontypes.AttributeChainID, chainID),
+			sdk.NewAttribute(clienttypes.AttributeKeyClientID, clientID),
+			sdk.NewAttribute(types.AttributeInitialHeight, clientState.LatestHeight.String()),
+			sdk.NewAttribute(
+				types.AttributeInitializationTimeout,
+				initTimeoutPeriod.String(),
+			),
+			sdk.NewAttribute(
+				types.AttributeTrustingPeriod,
+				clientState.TrustingPeriod.String(),
+			),
+			sdk.NewAttribute(
+				types.AttributeUnbondingPeriod,
+				clientState.UnbondingPeriod.String(),
+			),
+		),
+	)
+
+	return nil
+}
+
+func (k Keeper) MakeSubscriberGenesis(
+	ctx sdk.Context, req types.RegisterSubscriberChainRequest,
+) (*commontypes.SubscriberGenesisState, []byte, error) {
+	params := k.GetParams(ctx)
+	chainID := req.ChainID
+	k.Logger(ctx).Info("Creating genesis state for subscriber chain", "chainID", chainID)
+	chainIDWithoutRevision := avstypes.ChainIDWithoutRevision(chainID)
+	coordinatorUnbondingPeriod := k.stakingKeeper.UnbondingTime(ctx)
+	// client state
+	clientState := params.TemplateClient
+	clientState.ChainId = chainID
+	clientState.LatestHeight = clienttypes.GetSelfHeight(ctx)
+	trustPeriod, err := commontypes.CalculateTrustPeriod(
+		coordinatorUnbondingPeriod, params.TrustingPeriodFraction,
+	)
+	if err != nil {
+		return nil, nil, errorsmod.Wrapf(
+			sdkerrors.ErrInvalidHeight,
+			"error %s calculating self trusting period for chain %s",
+			err, chainID,
+		)
+	}
+	clientState.TrustingPeriod = trustPeriod
+	clientState.UnbondingPeriod = coordinatorUnbondingPeriod
+	// consensus state
+	consState, err := k.clientKeeper.GetSelfConsensusState(ctx, clientState.LatestHeight)
+	if err != nil {
+		return nil, nil, errorsmod.Wrapf(
+			clienttypes.ErrConsensusStateNotFound,
+			"error %s getting self consensus state for chain %s",
+			err, chainID,
+		)
+	}
+	operators, keys := k.operatorKeeper.GetActiveOperatorsForChainID(ctx, chainIDWithoutRevision)
+	powers, err := k.operatorKeeper.GetVotePowerForChainID(
+		ctx, operators, chainIDWithoutRevision,
+	)
+	if err != nil {
+		k.Logger(ctx).Error("error getting vote power for chain", "error", err)
```

Comment:
is the _**err**_ include "chainIDWithoutRevision" info ?

Reply:
Yes; added a comment to this effect just above.